### PR TITLE
[TT-3970] Override with deprecated auth when jwt or auth token is enabled

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This is an old hidden bug appeared with OAS.

### CODE REVIEW

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -875,20 +875,20 @@ func (a *APIDefinition) DecodeFromDB() {
...
-	makeCompatible := func(authType string) {
+	makeCompatible := func(authType string, enabled bool) {
 		if a.AuthConfigs == nil {
 			a.AuthConfigs = make(map[string]AuthConfig)
 		}
 
 		_, ok := a.AuthConfigs[authType]
 
-		if !ok {
+		if !ok && enabled {
```
Comment (vaske) [inline apidef/api_definitions.go:885]: do we need to cover case if it's not ok and enabled is false? or w are good like this?

**Review by vaske — APPROVED** [review]
LGTM, I left one minor comment but beside of that all is good

**Review by jeffy-mathew — APPROVED** [review]

**Pull request merged**